ond.py

```diff
@@ -61,59 +104,54 @@ def run_protocol(self, config_):
         )
 
         for test_id in config["test_ids"]:
+            algo_test_params = {
+                "dataset_ids": [],
+                "dataset_root": config["dataset_root"],
+                "image_features": {},
+            }
+
             test_metadata = self.interface.get_test_metadata(
                 session_id=session_id,
                 test_id=test_id
             )
 
-            test_params = {}
-
-            red_light = test_metadata.get("red_light", "")
-
-            # Assume no feedback_params attribute for now.
-
-            algorithm.execute("Initialize", config["detector_config"])
+            algo_test_data = {}
+            if "red_light" in test_metadata:
+                algo_test_data["red_light_image"] = test_metadata["red_light"]
 
-            test_params["image_features"] = {}
+            algorithm.initialize(algo_config_params)
 
             round_id = 0
             end_of_dataset = False
-
             while not end_of_dataset:
                 logging.info(f"Beginning round {round_id}")
-                file_list = self.interface.dataset_request(
-                    session_id, test_id, round_id
-                )
+                algo_test_data["round_id"] = round_id
 
-                if file_list is not None:
-                    features_dict, logits_dict = algorithm.execute(
-                        "FeatureExtraction", file_list
+                try:
```

Comment:
Should this be moved in round too? This would require changing the return type of `_run_round` to bool